### PR TITLE
make secretRef field optional

### PR DESCRIPTION
**Pull request by mitchdraft**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/864

### CODE REVIEW

File: projects/gloo/pkg/plugins/aws/ec2/credential_batched_instances.go [hunk 1]
```diff
@@ -141,6 +144,11 @@ func upstreamInstanceToEndpoint(writeNamespace string, upstream *v1.Upstream, in
 			Namespace: writeNamespace,
 		},
 	}
+	contextutils.LoggerFrom(ctx).Debugw("instance from upstream",
+		zap.Any("upstream", upstream),
```
Comment (yuval-k) [inline projects/gloo/pkg/plugins/aws/ec2/credential_batched_instances.go]: nit: `zap.Any("upstream", upstream)` -> `"upstream", upstream`
same for below

File: test/e2e/aws_ec2_test.go [hunk 2]
```diff
@@ -119,12 +129,28 @@ var _ = Describe("AWS EC2 Plugin utils test", func() {
 			return string(body), nil
 		}, "10s", "1s").Should(ContainSubstring(substring))
 	}
+
+	validateEc2Endpoint := func(envoyPort uint32, substring string) {
+		// first make sure that the instance is ready (to avoid false negatives)
+		By("verifying instance is ready - if this failed, you may need to restart the EC2 instance")
+		// stitch the url together to avoid bot spam
+		ec2Port := 80
+		ec2Url := fmt.Sprintf("http://%v:%v/metrics", strings.Join([]string{"52", "91", "199", "115"}, "."), ec2Port)
```
Comment (yuval-k) [inline test/e2e/aws_ec2_test.go]: what is this ip?

File: projects/gloo/pkg/plugins/aws/ec2/credential_batched_instances.go (hunk 1 above)
Comment (rickducott) [reply projects/gloo/pkg/plugins/aws/ec2/credential_batched_instances.go]: Elsewhere we’ve been using zap fields, so I prefer this just for consistency

**Comment by mitchdraft** [comment]
wip: need to add region config for env-derived sessions

File: test/e2e/aws_ec2_test.go (hunk 2 above)
Comment (mitchdraft) [reply test/e2e/aws_ec2_test.go]: that's a dummy ec2 instance for testing

**Comment by mitchdraft** [comment]
(no longer a wip) - added region config and tests using local credentials @yuval-k @rickducott

**Pull request merged**